compiler/rustc_mir/src/interpret/terminator.rs

```diff
@@ -425,6 +433,9 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 // recurse with concrete function
                 self.eval_fn_call(drop_fn, caller_abi, &args, ret, unwind)
             }
+            ty::InstanceDef::ErasedShim(_) => {
+                bug!("Interpreting type-erased functions is not implemented.")
```

Comment:
If this actually lands, I'd appreciate help making this code work in Miri. :)
Also, if the CTFE engine does not support these calls, we must make sure they are rejected in `const fn`.

Reply:
If this ever lands, these calls will only be created by a MIR-opt pass. As a consequence, they should not appear in a `const fn`.

Reply:
Even optimized code must, IMO, be executable by the Miri engine -- and indeed it is possible to run Miri on optimized MIR with `-Zmir-opt-level=4`. The code here is relevant well beyond `const fn`. So while it is okay to temporarily land a feature without Miri engine support, this is always a bug that should be fixed eventually.

Reply:
But I guess this means we indeed don't have to reject anything in `const fn`. However, we only recently stopped optimizing the MIR used by CTFE, and we might revert that decision at some point, so there's a danger of some lock-in here.